src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -531,6 +531,26 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithTemplateAnd
         return ResponseEntity.ok(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises/:exerciseId/with-auxiliary-repository/:auxiliaryRepositoryId
+     *
+     * @param exerciseId            the id of the programmingExercise to retrieve
+     * @param auxiliaryRepositoryId the id of the auxiliary repository of the exercise
+     * @return the ResponseEntity with status 200 (OK) and the programming exercise with template and solution participation, or with status 404 (Not Found)
+     */
+    @GetMapping("programming-exercises/{exerciseId}/with-auxiliary-repository/{auxiliaryRepositoryId}")
+    @EnforceAtLeastTutorInExercise
+    public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithAuxiliaryRepository(@PathVariable long exerciseId, @PathVariable long auxiliaryRepositoryId) {
```

Comment:
I'd assume that this endpoint only returns the exercise with the requested auxiliary repository but apparently all related aux repos are returned? In this case, why do you include the aux repo id at all?

Reply:
Good point, that's indeed redundant here. I removed th erepository Id here.